store/src/leaf_set.rs

```diff
@@ -199,4 +199,9 @@ impl LeafSet {
 	pub fn is_empty(&self) -> bool {
 		self.len() == 0
 	}
+
+	/// Iterator over positionns in the leaf_set (all leaf positions).
+	pub fn iter<'a>(&'a self) -> Box<Iterator<Item = u64> + 'a> {
```

Comment:
You can probably avoid the box with an impl Iterator?

Reply:
There was a problem somewhere when I attempted to do this but I did not look into it too closely (just added a `Box<>`...). Let me take another look and see if I can work out what was going on.

Reply:
Got it working. Still need the `Box<>` for `leaf_pos_iter()` on the Backend trait but `impl Iterator` works here.